analyzer/restapi-plugin/src/main/java/eu/fasten/analyzer/restapiplugin/LazyIngestionProvider.java

```diff
@@ -43,47 +48,93 @@ private static String toKey(String packageName, String version) {
     }
 
     public static void ingestArtifactIfNecessary(String packageName, String version, String artifactRepo, Long date) throws IllegalArgumentException, IOException {
-        var groupId = packageName.split(Constants.mvnCoordinateSeparator)[0];
-        var artifactId = packageName.split(Constants.mvnCoordinateSeparator)[1];
-        if(artifactRepo == null || artifactRepo.isEmpty()) {
-        	artifactRepo = MavenUtilities.MAVEN_CENTRAL_REPO;
-        }
-        if (!MavenUtilities.mavenArtifactExists(groupId, artifactId, version, artifactRepo)) {
-            throw new IllegalArgumentException("Maven artifact '" + packageName + ":" + version
-                    + "' could not be found in the repository of '"
-                    + (artifactRepo == null ? MavenUtilities.MAVEN_CENTRAL_REPO : artifactRepo) + "'."
-                    + " Make sure the Maven coordinate and repository are correct");
-        }
-        if (!hasArtifactBeenIngested(packageName, version)) {
-            var jsonRecord = new JSONObject();
-            jsonRecord.put("groupId", groupId);
-            jsonRecord.put("artifactId", artifactId);
-            jsonRecord.put("version", version);
-            jsonRecord.put("artifactRepository", artifactRepo);
-            if (date != null && date > 0) {
-            	// TODO why is the date necessary?
-                jsonRecord.put("date", date);
+        switch(KnowledgeBaseConnector.forge){
```

Comment:
As Magiel pointed out, please create helper methods for the logic inside the switch cases to reduce the complexity of the `ingestArtifactIfNecessary` method.

Reply:
Fixed on [28d46c8](https://github.com/fasten-project/fasten/pull/470/commits/28d46c85d7de232b6ede4bd34aa57fdaaa3e02da)